src/Tickets/Emails/Email_Abstract.php

```diff
@@ -828,4 +828,101 @@ public function get_post_type_data(): array {
 
 		return $data;
 	}
+
+	/**
+	 * Get a successful order array for preview.
+	 * 
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_preview_order_successful(): array {
+		return [
+			'provider'   => 'Stripe',
+			'status'     => 'success',
+			'post_title' => 'Black Midi with Special Guests Chat Pile and Apprehend',
+			'total'      => '$100.00',
+			'purchaser'  => [
+				'name' => 'John Doe',
+				'email' => 'john@doe.com',
+			],
+			'created' => 'March 1, 2023',
+			'id'      => 123,
+		];
+	}
+
+	/**
+	 * Get a failed order array for preview.
+	 * 
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_preview_order_failed(): array {
```

Comment:
Anything order related should be outside of the abstract class.

Reply:
I moved all preview data into its own class called `Preview_Data`. Any of the emails can fetch the data from there.